RiotSwiftUI/Modules/Onboarding/SplashScreen/View/OnboardingSplashScreen.swift

```diff
@@ -0,0 +1,200 @@
+    
+    /// Starts the animation timer for an automatic carousel effect.
+    private func startTimer() {
+        guard pageTimer == nil else { return }
+        
+        pageTimer = Timer.scheduledTimer(withTimeInterval: 5, repeats: true) { timer in
+            if viewModel.pageIndex == pageCount - 1 {
+                viewModel.send(viewAction: .hiddenPage)
+                
+                withAnimation(.easeInOut(duration: 0.7)) {
+                    viewModel.send(viewAction: .nextPage)
+                }
+            } else {
+                withAnimation(.easeInOut(duration: 0.7)) {
+                    viewModel.send(viewAction: .nextPage)
+                }
+            }
+        }
+    }
+    
+    /// Stops the animation timer for manual interaction.
+    private func stopTimer() {
+        guard let pageTimer = pageTimer else { return }
+        
+        self.pageTimer = nil
+        pageTimer.invalidate()
+    }
+    
+    // MARK: - Gestures
+    
+    /// Whether or not a drag gesture is valid or not.
+    /// - Parameter width: The gesture's translation width.
+    /// - Returns: `true` if there is another page to drag to.
+    private func shouldSwipeForTranslation(_ width: CGFloat) -> Bool {
+        if viewModel.pageIndex == 0  {
+            return isLeftToRight ? width < 0 : width > 0
+        } else if viewModel.pageIndex == pageCount - 1 {
+            return isLeftToRight ? width > 0 : width < 0
+        }
+        
+        return true
+    }
+    
+    /// Updates the `dragOffset` based on the gesture's value.
+    /// - Parameter drag: The drag gesture value to handle.
+    private func handleDragGestureChange(_ drag: DragGesture.Value) {
+        guard shouldSwipeForTranslation(drag.translation.width) else { return }
+        
+        stopTimer()
+        
+        // Animate the change over a few frames to smooth out any stuttering.
+        withAnimation(.linear(duration: 0.05)) {
+            dragOffset = isLeftToRight ? drag.translation.width : -drag.translation.width
+        }
+    }
+    
+    /// Clears the drag offset and informs the view model to switch to another page if necessary.
+    /// - Parameter viewSize: The size of the view in which the gesture took place.
+    private func handleDragGestureEnded(viewSize: CGSize) {
+        withAnimation(.easeInOut(duration: 0.2)) {
+            if dragOffset < -viewSize.width / 3 {
+                viewModel.send(viewAction: .nextPage)
+            } else if dragOffset > viewSize.width / 3 {
+                viewModel.send(viewAction: .previousPage)
+            }
+            
+            dragOffset = 0
+        }
+    }
+}
+
+// MARK: - Previews
+
+@available(iOS 14.0, *)
+struct OnboardingSplashScreen_Previews: PreviewProvider {
+    static let stateRenderer = MockOnboardingSplashScreenScreenState.stateRenderer
+    static var previews: some View {
+        stateRenderer.screenGroup()
+//            .environment(\.layoutDirection, .rightToLeft)
```

Comment:
I imagine it's intentional to keep this commented code?

Reply:
Ah probably not. I was using it whilst testing the layout 👍